src/shop.c

```diff
@@ -1159,11 +1159,26 @@ static void Task_ReturnToItemListAfterItemPurchase(u8 taskId)
 
     if (JOY_NEW(A_BUTTON | B_BUTTON))
     {
-        PlaySE(SE_SELECT);
+        u16 premierBallsToAdd = tItemCount / 10;
+        if (premierBallsToAdd >= 1
+            && ((I_PREMIER_BALL_BONUS <= GEN_6 && tItemId == ITEM_POKE_BALL)
+             || (I_PREMIER_BALL_BONUS >= GEN_7 && (ItemId_GetPocket(tItemId) == POCKET_POKE_BALLS))))
```

Comment:
If you mean moves, the main reason is that their PLA effects mostly apply to only that game's mechanics. Everything else, afaik, a feature introduced in LGPE is taken as GEN_7+

Reply:
I'm fairly sure there's exceptions to that (and I would personally disagree with that opinion since for me Gen7+ is primarily Alola and the main-games-but-not-really LGPE), but that's a conversation to be had in Discord at another time